extensions/json-schema/stac-eo-extension.json

```diff
@@ -0,0 +1,124 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "id": "stac-eo-spec.json#",
+  "title": "EO Item",
+  "type": "object",
+  "description": "STAC Earth Observation (EO) Extension to a STAC Item.  See https://github.com/radiantearth/stac-spec/blob/dev/extensions/stac-eo-spec.md",
+  "additionalProperties": true,
+  "allOf": [
+    {
+      "$ref": "#/definitions/eo"
+    }
+  ],
+  "definitions": {
+    "eo": {
+      "allOf": [
+        {
+          "oneOf": [
+            {
+              "$ref": "https://raw.githubusercontent.com/radiantearth/stac-spec/master/json-spec/json-schema/stac-item.json#/definitions/core"
```

Comment:
The validator (ajv) we are using in this repository doesn't allow referencing schemas by URL, so we couldn't use this in the CI. Can this be changed to a relative URL? Otherwise I'd just remove the inheritance. Don't see a major advantage here. You could just validate against Item an EO extension separately.

Reply:
If we use relative references then I think we will need all the JSON schemas to live in the same path, i.e. we able to directly construct the path between the relative references.  We can do this but we'd have to change the layout to put all JSON schema in one path (like all in one folder), but then you'd either need to put your custom extension schemas there or everybody who uses your custom schema would need to build the appropriate reference resolver.  I really put in the absolute reference to get out of the need for custom reference resolvers. 

The advantage I see with this inheritance is that we don't need to know the relationship between the various schemas in order to perform the validation.  The idea here is that the schema itself says that this extension is extending these various schemas and adding in the specified components.

For instance you could write a schema that requires the item to be both a EO item and a item with the start/stop time extension, plus some other custom fields...  The schema would include these various definitions from the extension's schemas and enforce this without the end validation user needing to know to check these other schemas.

Reply:
I understand all your points made here, but it just won't work with the validator we are using (see https://github.com/jessedc/ajv-cli/issues/29 ) and therefore we can't use it in CI. We either need to use another validator or remove the dependency to items for now.